Fixes #3277 - using assembly (#5368)
MicrosoftDocs/PowerShell-Docs: reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md

---
keywords: powershell,cmdlet
locale: en-us
ms.date: 01/29/2020
online version: https://docs.microsoft.com/powershell/module/microsoft.powershell.core/about/about_using?view=powershell-5.1&WT.mc_id=ps-gethelp
schema: 2.0.0
title: about_Using
---
# About Using

## SHORT DESCRIPTION
Allows you to indicate which namespaces are used in the session.

## LONG DESCRIPTION

The `using` statement allows you to specify which namespaces are used in the
session. Adding namespaces simplifies usage of .NET classes and member and
allows you to import classes from script modules and assemblies.

> The `using` statement needs to be the first statement in the script.

Reviewer: I'd change this to something like:

> `using` statements must come before any other statements in a script.

Author: fixed

The `using` statement should not be confused with the `using:` scope modifier
for variables. For more information, see
[about_Remote_Variables](about_Remote_Variables.md).

## Syntax

> To preload typenames from a .NET namespace:

Reviewer: I wouldn't say this preloads anything. It just adds prefixes to the type resolution search. See https://github.com/PowerShell/PowerShell/blob/148124e0551df813a5b57f071a5b826d7f44927c/src/System.Management.Automation/engine/parser/TypeResolver.cs#L335-L365.

Maybe something like:

> To specify .NET namespaces from which to resolve types

Author: Fixed

```
using namespace <.NET-namespace>
```

To load classes from a PowerShell module:

```
using module <module-name>
```

To preload classes from a .NET assembly:

```
using assembly <.NET-assembly-path>
using assembly <.NET-namespace>
```

> Loading a namespace makes it easier to reference types by their short names.

Reviewer: This isn't really a form of loading. Just adds places to look for types.

Author: Fixed

> Loading an assembly preloads classes into a script at parse time. This allows

Reviewer: ```suggestion
Loading an assembly preloads .NET types from that assembly into a script at parse time. This allows
```

Author: Fixed

> you to create new PowerShell classes that are derived from the preloaded

Reviewer: ```suggestion
you to create new PowerShell classes that use types from the preloaded assembly.
```

Author: Fixed

> classes. In Windows PowerShell 5.1 you can load the assembly by path name or by

Reviewer: ```suggestion
In Windows PowerShell 5.1 you can load the assembly by path name or by
```

Author: Fixed

> namespace. When you use the namespace name, PowerShell searches the .NET Global

Reviewer: ```suggestion
name. When you use the name, PowerShell searches the .NET Global
```

Author: Fixed

> Assembly Cache (GAC) for the associated DLL.

Reviewer: ```suggestion
Assembly Cache (GAC) for the associated assembly.
```

Author: Fixed

$hashfromstream.Hash.ToString()
```

### Example 2 - Load classes from a script module

> In this example, we have a PowerShell script module named **CardGames** the

Reviewer 2: the last `the` should be `that`.

Author: Fixed

defines the following classes:

- **CardGames.Deck**
- **CardGames.Card**

The `using module` command imports the module. `Import-Module` and the
`#requires` statement only import the module functions, aliases, and variables,
as defined by the module. Classes are not imported. The `using module`

> statement imports the classes defined in the module.

Reviewer 2: Can reword this? I think just the first sentence could be removed and it'll be clearer.

Author: The sentence is needed but it could be clearer.
